src/mds/journal.cc

```diff
@@ -140,7 +140,18 @@ void LogSegment::try_to_expire(MDSRank *mds, MDSGatherBuilder &gather_bld, int o
   for (elist<CInode*>::iterator p = dirty_dirfrag_nest.begin(); !p.end(); ++p) {
     CInode *in = *p;
     dout(10) << "try_to_expire waiting for nest flush on " << *in << dendl;
-    mds->locker->scatter_nudge(&in->nestlock, gather_bld.new_sub());
+    mds->locker->mark_nudging_nestlock(utime_t(), &in->nestlock, true);
+    C_MDS_NestlockNudged* fin = new C_MDS_NestlockNudged(mds, &in->nestlock, gather_bld.new_sub());
```

Comment:
the code introduces at least 2 memory allocation for each Locker::scatter_nudge() call. This is bad. I'd to avoid the overhead when no one propagate rstat.